falcon/api_helpers.py

```diff
@@ -126,12 +124,11 @@ def default_serialize_error(req, resp, exception):
 
     if preferred is not None:
         if preferred == 'application/json':
-            representation = exception.to_json()
+            resp.body = exception.to_json()
+            resp.content_type = preferred
         else:
-            representation = exception.to_xml()
-
-        resp.body = representation
-        resp.content_type = preferred + '; charset=UTF-8'
+            resp.body = exception.to_xml()
+            resp.content_type = preferred + '; charset=UTF-8'
```

Comment:
As I commented earlier, I think we can actually remove the charset parameter in this case as well. But we can just merge this PR as-is and then I'll rebase #1476 to suit.